### PR TITLE
Update the README of the text classification example

### PR DESCRIPTION
**Pull request by sgugger**
# What does this PR do?

This PR updates the main README of the examples folder and the one in the text classification example to take into account the recent changes in the scripts. In particular, I re-ran the command shown for all tasks with/without FP16 to make a clean table of results.

I moved all stuff about distributed training/TPUs in the general README of the examples as all example scripts now use Trainer, so have this working out of the box.

### CODE REVIEW

File: examples/README.md [hunk 1]
```diff
@@ -69,34 +69,78 @@ Coming soon!
 **Coming soon!**
 -->
 
+## Distributed training and mixed precision
+
+All the PyTorch scripts mentioned above work out of the box with distributed training and mixed precision, thanks to
+the [Trainer API](https://huggingface.co/transformers/main_classes/trainer.html). To launch one of them on 8 GPUS, use
```
Comment (patrickvonplaten) [inline examples/README.md:75]:
```suggestion
the [Trainer API](https://huggingface.co/transformers/main_classes/trainer.html). To launch one of them on the number of GPUS you have, use
```

or maybe `n` GPUS because below we are using `number_of_gpu_you_have` in the command

File: examples/text-classification/README.md [hunk 2]
```diff
@@ -14,7 +14,76 @@ See the License for the specific language governing permissions and
...
+```
+
+where task name can be one of cola, sst2, mrpc, stsb, qqp, mnli, qnli, rte, wnli.
+
+We get the following results on the dev set of the benchmark with the previous commands (with an exception for MRPC and
+WNLI which are tiny and where we used 5 epochs isntead of 3). Trainings are seeded so you should obtain the same
+results with PyTorch 1.6.0 (and close results with different versions), training times are given for information (a
+single Titan RTX was used):
+
+| Task  | Metric                       | Result      | Training time |
```
Comment (patrickvonplaten) [inline examples/text-classification/README.md:52]: this is awesome! Super useful

**Review by patrickvonplaten — APPROVED** [review]
Great! Have one nit in the beginning of the description
